scalableminds/webknossos
# Make far plane of td viewport adaptive to dataset

This PR makes the far clipping plane of the 3d port adaptive to the dataset extent in nm. This should prevent errors of like parts or all planes being clipped away in the 3d viewport. An example of such a problem can be found in #3849.

### URL of deployed dev instance (used for testing):
- https://adaptivefarplanefortdviewport.webknossos.xyz

### Steps to test:
- Download this [MPRAGE_250um dataset](https://static.webknossos.org/data/MPRAGE_250um.zip) and import it into wk (locally or testing instance).
- If you view it on the current master like running on wk.org (see [link](https://webknossos.org/datasets/Demo_Organization/MPRAGE_250um/view#320,440,440,0,1.30)) the 3d Viewport should be broken.
- Now view the dataset with the changes of this branch (locally or testing instance). The 3d viewport should now actually render all planes as expected.
- Play around with the rotation and the rotation buttons of the 3d viewport and try to find any position in which clipping still kicks in and the planes are not fully rendered. There should not be a case like this. At least I could not find any case like this during my testing.
- Try the same things on a "normal" dataset with a resolution in the nanometer range. The 3d viewport should also work here.

### Issues:
- fixes #3849

------
(Please delete unneeded items, merge only when none are left open)
